test_conformance/subgroups/test_workgroup.cpp

```diff
@@ -811,88 +817,38 @@ cl_uint set_bit(cl_uint bit_value, cl_uint number, cl_uint position) {
     number ^= (-(bit_value) ^ number) & (1UL << position);
     return number;
 }
-cl_uint4 generate_bit_mask(cl_uint subgroup_local_id, std::string mask_type) {
+cl_uint4 generate_bit_mask(cl_uint subgroup_local_id, std::string mask_type, cl_uint max_sub_group_size) {
+    typedef std::bitset<128> bs128;
+    bs128 mask128;
     cl_uint4 mask;
-    cl_uint first_el = 0;
-    cl_uint second_el = 0;
-    cl_uint third_el = 0;
-    cl_uint fourth_el = 0;
-    cl_uint pos = subgroup_local_id % 32;
-    if (mask_type == "eq") {
-        if (subgroup_local_id < 32) {
-            first_el = set_bit(1, 0, pos);
-            second_el = 0xFFFFFFFF;
-            third_el = 0xFFFFFFFF;
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (32 <= subgroup_local_id && subgroup_local_id < 64) {
-            second_el = set_bit(1, 0, pos);
-            third_el = 0xFFFFFFFF;
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (64 <= subgroup_local_id && subgroup_local_id < 96) {
-            third_el = set_bit(1, 0, pos);
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (96 <= subgroup_local_id && subgroup_local_id < 128) {
-            fourth_el = set_bit(1, 0, pos);
-        }
-    }
+    cl_uint pos = subgroup_local_id;
+    if (mask_type == "eq")
+        mask128.set(pos);
     if (mask_type == "le" || mask_type == "lt") {
         cl_uint value = 0;
-        for (int i = 0; i <= pos; i++) {
-            value = set_bit(1, value, i);
-        }
-        if (mask_type == "lt") {
-            value = set_bit(0, value, pos);
-        }
-        if (subgroup_local_id <= 32) {
-            first_el = value;
-            second_el = 0xFFFFFFFF;
-            third_el = 0xFFFFFFFF;
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (32 < subgroup_local_id && subgroup_local_id <= 64) {
-            second_el = value;
-            third_el = 0xFFFFFFFF;
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (64 < subgroup_local_id && subgroup_local_id <= 96) {
-            third_el = value;
-            fourth_el = 0xFFFFFFFF;
-        }
-        if (96 < subgroup_local_id && subgroup_local_id <= 128) {
-            fourth_el = value;
-        }
+        for (cl_uint i = 0; i <= pos; i++)
+            mask128.set(i);
+        if (mask_type == "lt")
+            mask128.reset(pos);
     }
     if (mask_type == "ge" || mask_type == "gt") {
         cl_uint value = 1;
```

Comment:
now unused ? The same with set_bit function ?

Reply:
Removed unused variabled. set_bit function is still used for ballot tests.